File: aoc-2015/src/test/java/com/capital7software/aoc2015/days/Day06Test.java
package com.capital7software.aoc2015.days;

import com.capital7software.aoc.aoc2015.days.Day06;
import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.logging.Logger;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day06Test extends AdventOfCodeTestBase {
    private static final Logger LOGGER = Logger.getLogger(Day06Test.class.getName());

    @BeforeEach
    void setUp() {
        var instance = new Day06();
        setupFromFile(instance.getDefaultInputFilename());
    }

    @Test
    void getLitLights() {
        var instance = new Day06();
        var expectedCount = 998_996;
        var lights = instance.loadLights(lines, 1_000, 1_000);
        lights.applyInstructions();
        var actualCount = instance.getOnLightCount(lights);

        assertEquals(expectedCount, actualCount);
    }

    @Test
    void getTotalBrightnessOfLights() {
        var instance = new Day06();
        var expectedCount = 1_001_996;
        var lights = instance.loadLights(lines, 1_000, 1_000);
        lights.applyNewInterpretationOfInstructions();
        var actualCount = instance.getTotalBrightness(lights);

        assertEquals(expectedCount, actualCount);
    }

    @Override
    protected Logger getLogger() {
        return LOGGER;
    }
}

File: aoc-2016/src/test/kotlin/com/capital7software/aoc/aoc2016aoc/days/Day06Test.kt
package com.capital7software.aoc.aoc2016aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day06Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day06Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day06()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testErrorCorrectSignal() {
    val instance = Day06()
    val expected = "easter"
    val actual = instance.errorCorrectSignal(lines)

    assertEquals(expected, actual, "The error corrected signal "
        + "is not what was expected: $expected")
  }

  @Test
  fun testLeastCommonErrorCorrectSignal() {
    val instance = Day06()
    val expected = "advent"
    val actual = instance.errorCorrectSignal(lines, true)

    assertEquals(expected, actual, "The error corrected signal "
        + "is not what was expected: $expected")
  }
}
